sklearn/cluster/_kmeans.py

```diff
@@ -1166,7 +1190,8 @@ def fit(self, X, y=None, sample_weight=None):
             accept_large_sparse=False,
         )
 
-        self._check_params(X)
+        self._check_params_vs_input(X)
```

Comment:
This is a second round of checks after data validation that deals with valid values that depend on the data or on other parameters.